src/decision/src/VisionDecision.cpp

```diff
@@ -84,73 +100,70 @@ int VisionDecision::getAngleOfLine(bool rightSide, double numSamples, const sens
 
     // Assign garbage values
     int toParseFrom = -1;
-    int topRow = -1;
+    int bottomRow = -1;
+    
+    // Initialize how and where to parse.
+    incrementer = initializeIncrementerPosition(rightSide, image_scan, &startingPos);
 
-    // Decides how to parse depending on if rightSide is true or false.
-    if (rightSide) {
-        // starts at right side then increments to the left
-        startingPos = image_scan->width - 1;
-        incrementer = -1;
-    } else {
-        // starts at left side then increments to th right
-        startingPos = 0;
-        incrementer = 1;
-    }
-
-    // starts parsing from the right and finds where the highest white line
+    // starts parsing from the right and finds where the lowest white line
     // begins.
-    for (int i = 0; i < imageHeight - 1; i++) {
+    for (int i = imageHeight - 1; i > 0; i--) {
         int startPixel = getEdgePixel(startingPos, incrementer, i, image_scan, 1);
-        if (startPixel != -1 && topRow == -1) {
-            // Once found makes note of the highest point of the white line
-            // and the column it starts at.
-            topRow = i;
+        if (startPixel != -1 && bottomRow == -1) {
+            bottomRow = i;
             toParseFrom = startPixel - 1;
         }
     }
 
-    // Each slope will be compared to the top point of the highest white line
-    double x1 = getMiddle(toParseFrom, topRow, rightSide, image_scan);
+    // Each slope will be compared to the bottom point of the lowest white line
+    double x1 = getMiddle(toParseFrom, bottomRow, rightSide, image_scan);
 
     double sumAngles = 0;
-
     // Finds slopes (corresponding to number of samples given) then returns the sum of all slopes
     // also counts how many of them are valid.
-    for (double division = 0; division < numSamples && (numSamples + topRow) < imageHeight; division++) {
-        double yCompared = numSamples + topRow;
+    for (double division = 30; (division < numSamples) && (bottomRow - division > 0); division++) {
```

Comment:
Fixed - uses division = 1 instead of division = 30. (It needed to start above the starting point so I arbitrarily chose 30 in the beginning but forgot to change it).